### PR TITLE
 add print button and put patient's details at top of visit history page #7270

### LINKED ISSUE #7270: bug: In Visit History, print, or save as PDF, a multi-page list of encounters doesn't print or save all results.
In OpenEMR 7.0.1 (1), in Visit History, for a patient where there are numerous pages of results, Ctrl+P and then either printing, or saving as PDF, in any web browser on Windows or Linux, gives inconsistent results, and never successfully prints or saves the full list of encounters for the patient.

### To Reproduce

Use either "Visit History" icon in the toolbar / banner area or the "Visit History" tab open when a user logs in.

Results per page: ALL

Ctrl+P
Pages: All

A multi-page web page on other web sites works as expected with Ctrl+P.  So it appears to us to be an OpenEMR issue.

### PR DESCRIPTION
**Pull request by ruthkonyn**
Fixes #7270

#### Short description of what this resolves:
to print or make a pdf of a visit history report one had to use the browsers print function, with ctrl+p or file/print etc. This only prints what had been displayed on the screen, so for a long history that takes several screens only the first would be printed.

#### Changes proposed in this pull request:
add a 'print' button that prints the complete page of the history being displayed. for page length 'ALL' this is the complete history which will produce a multi page document where necessary
display the patient's details at the top of the history window and on the first page of the print or pdf document.

### CODE REVIEW

File: interface/patient_file/history/encounters.php [hunk 1]
```diff
@@ -401,7 +422,7 @@ function changePageSize() {
 
             $countRes = sqlStatement($countQuery, $sqlBindArray);
             $count = sqlFetchArray($countRes);
-            $numRes = $count['c'];
+            $numRes += $count['c'];
```
Comment (adunsulag) [inline interface/patient_file/history/encounters.php:425]: It looks like this didn't actually change the code logic, was thinking it did, but on further examination it keeps the code the same.

File: interface/patient_file/history/encounters.php [hunk 2]
```diff
@@ -297,6 +308,16 @@ function changePageSize() {
     </span>
 
     <br />
+    <span class="heading" >
+    <?php
+  // RM put patienes name, id and dob at top of the history -->
```
Comment (adunsulag) [inline interface/patient_file/history/encounters.php:313]: This breaks if the attendant_type is not a patient (IE therapy group encounters are being viewed.  We need to surround all of this:
```
 if ($attendant_type == 'pid') {
```

File: interface/patient_file/history/encounters.php (hunk 1 above)
Comment (ruthkonyn) [reply interface/patient_file/history/encounters.php:425]: oh yes, apologies: - i've been working on this code to sort out the problems where the documents are displayed on every page, issue #7275, and this new line has crept in from that work. It's not needed simply to add the print functionality. I'll fix and resubmit with a new PR

**Comment by ruthkonyn** [comment]
thank you so much stephen - i was just fixing these for a new PR when i noticed you'd done it already :-)

File: interface/patient_file/history/encounters.php [hunk 3]
```diff
@@ -297,6 +307,17 @@ function changePageSize() {
...
 
     <br />
+    <span class="heading" >
+    <?php
+    if ($attendant_type == 'pid') {
+  // RM put patienes name, id and dob at top of the history -->
+        $name =  getPatientNameFirstLast($pid);
+        $dob =  text(oeFormatShortDate(getPatientData($pid, "DOB")['DOB']));
+         $external_id = getPatientData($pid, "pubpid")['pubpid'];
+        echo $name . " (" . $external_id . ")" .  "&nbsp;  &nbsp; DOB: " . $dob ;
```
Comment (bradymiller) [inline interface/patient_file/history/encounters.php:317]: need escaping:
```
echo text($name ). " (" . text($external_id) . ")" .  "&nbsp;  &nbsp; DOB: " . $dob ;
```

**Pull request merged**